src/GraphQL/Resolver/AssetListing.php

```diff
@@ -0,0 +1,253 @@
+use Pimcore\Bundle\DataHubBundle\GraphQL\Exception\ClientSafeException;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Exception\NotAllowedException;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Service;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Traits\ServiceTrait;
+use Pimcore\Bundle\DataHubBundle\PimcoreDataHubBundle;
+use Pimcore\Bundle\DataHubBundle\WorkspaceHelper;
+use Pimcore\Db;
+use Pimcore\Model\Asset;
+use Pimcore\Model\Listing\AbstractListing;
+use Pimcore\Bundle\DataHubBundle\GraphQL\Helper;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class AssetListing
+{
+    use ServiceTrait;
+
+    protected $eventDispatcher;
+
+    /**
+     * AssetListing constructor.
+     * @param Service                  $graphQlService
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(Service $graphQlService, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->setGraphQLService($graphQlService);
+
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return mixed
+     */
+    public function resolveEdge($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        $element = $value['node'];
+
+        if (null !== $element) {
+            return $this->extractSingleElement($element, $args, $context, $resolveInfo);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param null $value
+     * @param array $args
+     * @param array $context
+     * @param ResolveInfo|null $resolveInfo
+     * @return array
+     * @throws \Exception
+     */
+    public function resolveListing($value = null, $args = [], $context, ResolveInfo $resolveInfo = null)
+    {
+        if ($args && isset($args['defaultLanguage'])) {
+            $this->getGraphQlService()->getLocaleService()->setLocale($args['defaultLanguage']);
+        }
+
+        $modelFactory = $this->getGraphQlService()->getModelFactory();
+        $listClass = Asset\Listing::class;
+
+        /** @var AbstractListing $objectList */
+        $objectList = $modelFactory->build($listClass);
+        $conditionParts = [];
+        if (isset($args['ids'])) {
+            $conditionParts[] = '(id IN (' . $args['ids'] . '))';
+        }
+
+        // paging
+        if (isset($args['first'])) {
+            $objectList->setLimit($args['first']);
+        }
+
+        if (isset($args['after'])) {
+            $objectList->setOffset($args['after']);
+        }
+
+        // sorting
+        if (!empty($args['sortBy'])) {
+            $objectList->setOrderKey($args['sortBy']);
+            if (!empty($args['sortOrder'])) {
+                $objectList->setOrder($args['sortOrder']);
+            }
+        }
+
+        // Include unpublished
+//        if (isset($args['published']) && $args['published'] === false) {
```

Comment:
this can be removed , no unpublished assets

Reply:
done